share/poly.go

```diff
@@ -426,3 +410,69 @@ func RecoverCommit(g kyber.Group, shares []*PubShare, t, n int) (kyber.Point, er
 
 	return Acc, nil
 }
+
+// RecoverPubPoly reconstructs the full public polynomial from a set of public
+// shares using Lagrange interpolation.
+func RecoverPubPoly(g kyber.Group, shares []*PubShare, t, n int) (*PubPoly, error) {
+	x := make(map[int]kyber.Scalar)
+	for i, s := range shares {
+		if s == nil || s.V == nil || s.I < 0 || n <= s.I {
+			continue
+		}
+		x[i] = g.Scalar().SetInt64(1 + int64(s.I))
+	}
+
+	if len(x) < t {
+		return nil, errors.New("share: not enough good public shares to reconstruct secret commitment")
+	}
+
+	var accPoly *PubPoly
+	var err error
+
+	for j := range x {
+		basis := LagrangeBasis(g, j, x)
+		// compute the L_j * y_j polynomial in point space
+		tmp := basis.Commit(shares[j].V)
+		if accPoly == nil {
+			accPoly = tmp
+			continue
+		}
```

Comment:
I prefer to read loops which initialize an accumulator when they are wrtitten with an else and no continue:
```
for {
  if acc == nil {
    acc = tmp
  } else {
   var err error
    acc, err = add.Add(tmp)
    if err != nil {
      return nil, err
    }
  }
} //end for
```

Reply:
Mhhhh. I understand your preference, mine is more along the lines of clarity of code: 
+ It's simpler / easier to read without an additional indentation and if/else case to mentally remember.
+ It clearly indicates a breaking condition in the loop, same as returning early from a function.  

For example, we never do this:
```go
if err := myDangerousOperation(); err != nil {
   return err
} else {
   mySafeOperation()
}
```
but simply
```go
if err := myDangerousOperation(); err != nil {
   return err
} 
mySafeOperation()
```
What's your reasonings for not liking `continue` ? C/C++ background ? ;)